src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/stdx/type_traits.hpp

```diff
@@ -0,0 +1,368 @@
+#pragma once
+
+#include <type_traits>
+
+#include <bsoncxx/config/prelude.hpp>
+
+namespace bsoncxx {
+
+BSONCXX_INLINE_NAMESPACE_BEGIN
+namespace stdx {
+inline namespace type_traits {
```

Comment:
Providing public API for type traits may add maintenance cost for API that is less related to the goals of bsoncxx library. I suggest discouraging consumers from using directly:

```suggestion
// API in the `type_traits_private` namespace is unstable. Do not use directly.
namespace type_traits_private {
```

Reply:
I tweaked the namespace layout to be `bsoncxx::_traits`, to keep it easy to type but also "obviously not public". Bike-shedding over the namespace name is TBD.